README.md

```diff
@@ -803,6 +807,62 @@ $ ulimit -u # processes / threads
 
 Just pass a new number as the argument to change it.
 
+## Prometheus Support
+
+Vegeta has a built-in Prometheus Exporter that may be enabled during "attacks" so that you can point any Prometheus instance to Vegeta instances and get some metrics about http requests performance and about the Vegeta process itself.
```

Comment:
While I like this for a lot reasons, I think vegeta should be exporting metrics to a Prometheus push gateway rather than being scraped directly by prometheus.

Since vegeta isn't a long lived process you'll have race conditions where Prometheus might not scrape the last bit of attack data before vegeta shuts down.

Prometheus [best practices docs](https://prometheus.io/docs/practices/pushing/) say you should use push gateway for "service-level batch jobs" which is what I think vegeta would qualify as:
> Usually, the only valid use case for the Pushgateway is for capturing the outcome of a service-level batch job. A "service-level" batch job is one which is not semantically related to a specific machine or job instance (for example, a batch job that deletes a number of users for an entire service). Such a job's metrics should not include a machine or instance label to decouple the lifecycle of specific machines or instances from the pushed metrics. This decreases the burden for managing stale metrics in the Pushgateway. See also the best practices for monitoring batch jobs.

Reply:
I bring this up, bc I think it should instead have, or at least additionally offer, a report type of `prom` which will then export metrics from a report to a prometheus push gateway.

I am currently writing that for `datadog` rather than Prometheus but the idea is the same. The advantage of the report is that one could take an existing Vegeta test result and push into a metrics store rather than having to run a new test.

And the user could publish the results to a Prometheus and a DataDog and wherever else they needed with the reporting decoupled from the attacking.

Reply:
Note to self: When `-prometheus-addr` is set and we start an HTTP server for prometheus to scrape, make sure that all metrics are scrapped before exiting from the program, even after the attack finished per se.